.github/workflows/run-unit-tests.yml

```diff
@@ -0,0 +1,66 @@
+name: Run unit tests
+
+on:
+  pull_request:
+    types: [opened,reopened,synchronize]
+
+jobs:
+  test-unit-jest:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 'lts/*'
+
+      - run: corepack enable
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: '.nvmrc'
+          cache: yarn
+
+      - name: Install dependencies
+        run: yarn --immutable
+
+      - name: test:coverage:jest
+        run: yarn test:coverage:jest
+
+      - uses: codecov/codecov-action@d9f34f8cd5cb3b3eb79b3e4b5dae3a16df499a70
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          fail_ci_if_error: true
+
+  test-unit-jest-development:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 'lts/*'
+
+      - run: corepack enable
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: '.nvmrc'
+          cache: yarn
+
+      - name: Install dependencies
+        run: yarn --immutable
```

Comment:
Is there a way to make these steps, or at least some of them (especially the `yarn --immutable` step), part of the same workflow so we don't have to repeat work?

Reply:
It should be possible. I will try to investigate this more. Ideally, there would be one setup-environment workflow, that would setup nodejs, enable corepack, checkout the repository, and install the dependencies. We could reuse this workflow for every other workflow, as this code is copy-pasted for every workflow currently.

Reply:
Implemented the setup-environment action (that should be ported to all the other workflows in the future), removed the repeated work.